### DISCUSSION TITLE
Navigation with Joyride

### DISCUSSION THREAD

**Discussion by ijdershem-jf** [gilbarbara/react-joyride#670] [Q&A] [answered]
I'm hoping to implement Joyride in my app but I'm having some difficulty figuring out how to handle step iteration after a navigation. Does anyone know of a good approach to waiting for navigation before incrementing the tour step?

**Comment by gilbarbara** [comment, marked as answer]
hey @ijdershem-jf 

You'll need to use the controlled mode. 
Please refer to the [demo](https://github.com/gilbarbara/react-joyride-demo/blob/master/src/Controlled/index.tsx) source code or this tutorial #666

**Reply by ijdershem-jf** [reply]
Great, thank you! I was able to use the controlled mode in combination with a setTimeOut() call to delay step iteration & tooltip rendering until navigation/component mount was complete.

**Reply by gilbarbara** [reply]
setTimeout is very, very fragile and will probably break for users with slow connections

**Reply by ijdershem-jf** [reply]
Do you have a good suggestion for checking that a target has mounted? Within a functional component that is

**Reply by gilbarbara** [reply]
componentDidMount ou a useEffect without dependencies.
